src/service/ProductManagementService.js

```diff
@@ -0,0 +1,162 @@
+import { CONFIRMATION_RESPONSES } from '../constants/constants.js';
+import getUserConfirmation from '../utils/getUserConfirmation.js';
+
+class ProductManagementService {
+  #stock;
+  #promotion;
+  #productsInfo;
+
+  constructor(stock, promotion) {
+    this.#stock = stock;
+    this.#promotion = promotion;
+    this.#productsInfo = [];
+  }
+
+  // 구매할 상품 및 수량 저장
+  initProductsInfo(productsInfo) {
+    this.#productsInfo = productsInfo;
+  }
+
+  // 구매할 상품 및 수량 전체 정보 반환
+  getProductsInfo() {
+    return this.#productsInfo;
```

Comment:
`getter`를 구현할 때 구조 분해 할당을 활용하면 외부에서 해당 필드에 대한 접근을 막을 수 있어서 캡슐화/은닉화에 더욱 유용합니당!
```suggestion
    return [ ...this.#productsInfo ];
```

Reply:
헉,, 이렇게 써볼 생각은 전혀 못했는데 좋은 정보 감사해요! 🤩 스프레드 연산자 쓰면 아무래도 새로운 배열 반환해주니까 불변성 보장해주는거군요